### PR TITLE
Adding icmp support

### PR DESCRIPTION
**Pull request by aimanparvaiz**
Attempt to add ICMP protocol handling while setting up AWS security group rules.

### CODE REVIEW

**Comment by aimanparvaiz** [comment]
I found another issue with ICMP PR and am working on it. Will update you once its done.

**Comment by aimanparvaiz** [comment]
I added a test for tcp protocol with port zero. All tests are passing.

File: README.md [hunk 1]
```diff
@@ -38,6 +40,7 @@ It should be easier to understand a valid configuration based on example:
 
       rules:
         - tcp port 80 0.0.0.0/0
+        - icmp port 8 192.168.1.1/32 # ICMP Type 13; Timestamp
```
Comment (rustyrazorblade) [inline README.md:43]: wouldn't this be type 8?

**Comment by rustyrazorblade** [comment]
merged

**Pull request closed without merging**